### PR TITLE
6th v1alpha1 session

### PR DESCRIPTION
**Pull request by davidhadas**
This is the last PR that includes DataTypes using the common structure. 

The Pod dataType included here is using /proc/net/* data to discover pod networking information. 
This requires parsing of /proc/net/* ipv4 and ipv6 data.
Once the information is parsed, the IpSet dataType is used to do the rest.

The Req and Resp data types included are constructed from previously merged dataTypes.

And last, the Session DataType, is constructed from req, Resp, Pod, and Body to describe a complete request-response session as well as the pod information during the session. 

This completes a set of 18 dataTypes presently included in the pkg/apis/wsecurity/v1alpha1 package which serves as the data processing foundation for guard-gate, guard learner, and guard-ui. 

The next PR will add code-generation of deepCopy and a K8s clientset to this package.

### CODE REVIEW

File: pkg/apis/wsecurity/v1alpha1/httpReq.go [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+			length >>= 1
+		}
+		profile.ContentLength = CountProfile(log2length)
+	}
+}
+
+//////////////////// ReqPile ////////////////
+
+// Exposes ValuePile interface
+type ReqPile struct {
```
Comment (psschwei) [inline pkg/apis/wsecurity/v1alpha1/httpReq.go:65]: Given that the `ReqProfile` and `ReqPile` types have the same elements, would there be any benefit to doing something like this:

```golang
type Req struct {
	ClientIp      IpSetProfile     `json:"cip"`           // 192.168.32.1
	HopIp         IpSetProfile     `json:"hopip"`         // 1.2.3.4
	Method        SetProfile       `json:"method"`        // GET
	Proto         SetProfile       `json:"proto"`         // "HTTP/1.1"
	MediaType     MediaTypeProfile `json:"mediatype"`     // "text/html"
	ContentLength CountProfile     `json:"contentlength"` // 0
	Url           UrlProfile       `json:"url"`
	Qs            QueryProfile     `json:"qs"`
	Headers       HeadersProfile   `json:"headers"`
}

type ReqProfile Req
type ReqPile Req
```

Comment (psschwei) [reply pkg/apis/wsecurity/v1alpha1/httpReq.go:65]: (realize we didn't do it for previous types, so it might require a larger refactor)

Comment (davidhadas) [reply pkg/apis/wsecurity/v1alpha1/httpReq.go:65]: not the same elements...
while ReqProfile is a structure of IpSetProfile , SetProfile , UrlProfile etc. 
ReqPile is a structure of IpSetPile, SetPile, UrlPile etc.

Comment (psschwei) [reply pkg/apis/wsecurity/v1alpha1/httpReq.go:65]: :facepalm:

**Review by psschwei — COMMENTED** [review]
/lgtm

**Pull request merged**